src/ts/lib/util/common.ts

```diff
@@ -945,6 +951,79 @@ class UtilCommon {
 		return String(s || '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
 	};
 
+	private getPrismComponentDependencies (lang: string) {
+		const components = require(`prismjs/components.js`);
+		const result = [];
+		const language = components.languages[lang] || {};
+		// the type of `require`, `optional`, `alias` is one of `undefined`, `string[]` and `string`
+		const requirements = [].concat(language.require || []);
+		const optionals = [].concat(language.optional || []);
+		requirements.concat(optionals).forEach(item => {
+			result.push(...this.getPrismComponentDependencies(item));
+		});
+		result.push(lang);
+		return result;
+	};
+
+	/** returns an array which is the correct order of loading all Prism components */
+	prismComponents (): string[] {
+		const components = require(`prismjs/components.js`);
+		const codes = components.languages || {};
+		const result = [];
+		Object.keys(codes).forEach((key: string) => {
+			codes[key].title && result.push(...this.getPrismComponentDependencies(key));
+		});
+		return [...new Set(result)];
+	};
+
+	findRealPrismLanguageCode (lang: string){
+		if (!this.prismMap[lang]) {
```

Comment:
I do not understand this method, it looks very slow, if you will have 100 code blocks on a page this logic becomes can become very slow. The idea behind creating a single map is to have the option to find lang title by code.